### ISSUE TITLE
Add support for files in luacheckrc

### ISSUE THREAD

**Issue by ghprince** [flycheck/flycheck#1046]
As of #767 , luacheck are reading contents from standard input, which means following config in .luacheckrc will not work

```lua
files['*_spec.lua'] = { std = '+busted' }
```

since luacheck won't know the file name.

Anyway to workaround this?

**Comment by lunaryorn** [comment]
@ghprince Maybe, I don't know lua. 

You may check the documentation of luacheck or ask its developers/maintainers whether we can somehow tell luacheck which file it's checking.  Some tools that we use have command line options for this purpose, for instance `--stdin-file-name` in eslint.  Maybe there's something similar for luacheck.  

Could you try and find out?

**Comment by ghprince** [comment]
@lunaryorn Thanks a lot for your quick response! I walked through luacheck doc and did find this:

> File can be passed through stdin using - as argument or using a temporary file. Real filename should be passed using --filename option.

I'll create a pull request for this :)

**Comment by lunaryorn** [comment]
Awesome, thank you! ☺️
